src/core/ComboBox/ComboBox.tsx

```diff
@@ -103,25 +119,44 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     [options, id],
   );
 
+  const userOnChange = React.useRef(onChange);
+
   const memoizedItems = React.useMemo(
     () =>
-      options.map(({ label, value, ...rest }, index) => (
-        <MenuItem
-          id={getOptionId(index)}
-          key={getOptionId(index)}
-          value={value}
-          role='option'
-          onClick={(value: T) => {
+      options.map((option, index) => {
+        const { label, value, ...rest } = option;
+        const additionalProps = {
+          value: value,
+          role: 'option',
+          onClick: () => {
             setSelectedValue(value);
-            onChange?.(value);
+            userOnChange.current?.(value);
             setIsOpen(false);
-          }}
-          {...rest}
-        >
-          {label}
-        </MenuItem>
-      )),
-    [options, getOptionId, onChange],
+          },
+        };
+        if (itemRenderer) {
+          return React.cloneElement(
+            itemRenderer(option, {
+              id: getOptionId(index),
+              index,
+              isSelected: false,
+              isFocused: false,
+            }),
+            additionalProps,
```

Comment:
How it will work?

Reply:
Not sure what you're asking. These additionalProps just get applied on top of the user's returned JSX

Reply:
Nvm 😀